bootstrap/secret/secure_test.go

```diff
@@ -419,3 +419,89 @@ func TestSecureProvider_ListSecretPathsSecrets(t *testing.T) {
 		})
 	}
 }
+
+func TestSecureProvider_SecretUpdatedAtPath(t *testing.T) {
+	mock := &mocks.SecretClient{}
+	tests := []struct {
+		Name     string
+		Config   TestConfig
+		Client   secrets.SecretClient
+		Path     string
+		Callback func(path string)
+	}{
+		{"Valid Secure", TestConfig{}, mock, expectedPath, func(path string) {}},
+		{"Valid No Callbacks", TestConfig{}, mock, expectedPath, nil},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			target := NewSecureProvider(context.Background(), tc.Config, logger.MockLogger{}, nil, nil, "testService")
+			target.SetClient(tc.Client)
+
+			if tc.Callback != nil {
+				target.registeredSecretCallbacks[tc.Path] = tc.Callback
+			}
+
+			target.SecretUpdatedAtPath(tc.Path)
+		})
+	}
+}
+
+func TestSecureProvider_RegisteredSecretUpdatedCallback(t *testing.T) {
+	mock := &mocks.SecretClient{}
+
+	tests := []struct {
+		Name     string
+		Config   TestConfig
+		Client   secrets.SecretClient
+		Path     string
+		Callback func(path string)
+	}{
+		{"Valid Secure", TestConfig{}, mock, expectedPath, func(path string) {}},
+		{"Valid No Callbacks", TestConfig{}, mock, expectedPath, nil},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			target := NewSecureProvider(context.Background(), tc.Config, logger.MockLogger{}, nil, nil, "testService")
+			target.SetClient(tc.Client)
```

Comment:
Is the mock really needed since you don't have an mock.On() calls? i.e. the client is never called by this test.